HW1/P5/P5.txt

```diff
@@ -1 +1,25 @@
 # Your discussion here
+---------Shortest Path----------------
+
+'Kevin_Bacon'--------->'John_Lithgow'--------->Harvard_University(arrived!)
+
+and
+
+'Harvard_University'--------->'September_16'--------->'Mickey_Rourke'--------->Kevin_Bacon(arrived!)
+
+--------------------------------------
+
+
+The above path is not unique but it is correct. Since I check 'John_Lithow' was a neighbor of 'Kevin_Bacon' and 'Harvard_University' is a neighbor of 'John_Lithow'. 
+
+'September_16' is a neighbor of 'Harvard_University', 'Mickey_Rourke' is a neighbor of 'September_16', and 'Kevin_Bacon' is a neighbor of 'Mickey_Rourke'.
+
+
+
+For connected components:
+My script is correctly working on P4's small dataset but not on the large dataset of P5.
+
+My algorithm is:
+1, make two different graphes - asymmetric and symmetric
+2, bfs on one note, find a cc, then change to another node which is not in that cc
```

Comment:
I don't think the scaling of this algorithm will allow it to work on large datasets. Since you use BFS repeatedly, I think the algorithm will converge in roughly O(N). You can write different algorithms that converge on O(D) or O(log[D]) where D is the diameter of the graph.